### PR TITLE
Refactor to use `From` trait for `String`

### LINKED ISSUE #170: Update `String` to use `From` trait over special functions
#### Description

The current implementation of `String` uses special functions for conversion between types. This should instead use the `From` trait.

#### Motivation

Multiple `From` implementations are now functioning as intended.

### PR DESCRIPTION
**Pull request by bitzoic**
## Type of change

- Improvement (refactoring, restructuring repository, cleaning tech debt, ...)

## Changes

The following changes have been made:

- Uses `From<Vec>` instead of `as_vec` and `from_utf8`
- Uses `From<raw_slice>` instead of `from_raw_slice` 

## Notes

- This is a breaking change

## Related Issues

Closes #170

### CODE REVIEW

File: libs/strings/string/src/lib.sw [hunk 1]
```diff
@@ -173,17 +149,33 @@ impl String {
...
+        asm(ptr: (self.bytes.buf.ptr(), self.bytes.len)) { ptr: raw_slice }
+    }
+}
+
+impl From<Vec<u8>> for String {
+    /// Converts a vector of bytes to a `String`.
+    ///
+    /// # Arguments
+    ///
+    /// * `bytes` - The vector of `u8` bytes which will be converted into a `String`.
```
Comment (K1-R1) [inline libs/strings/string/src/lib.sw:169]: This doesn't match the `from` function below

**Review by K1-R1 — CHANGES_REQUESTED** [review]

File: libs/strings/string/src/lib.sw [hunk 2]
```diff
@@ -173,17 +149,33 @@ impl String {
...
-//             bytes
-//         }
-//     }
-
-//     fn into(self) -> raw_slice {
-//         asm(ptr: (self.bytes.buf.ptr(), self.bytes.len)) { ptr: raw_slice }
-//     }
-// }
+impl From<raw_slice> for String {
+    fn from(slice: raw_slice) -> String {
```
Comment (Braqzen) [inline libs/strings/string/src/lib.sw:153]:
```suggestion
    fn from(slice: raw_slice) -> Self {
```

File: libs/strings/string/src/lib.sw [hunk 3]
```diff
@@ -173,17 +149,33 @@ impl String {
...
+        Self {
+            bytes: Bytes::from_raw_slice(slice),
```
Comment (Braqzen) [inline libs/strings/string/src/lib.sw:155]: I assume this will be updated to use the `from` on `Bytes`

Comment (bitzoic) [reply libs/strings/string/src/lib.sw:155]: Yes a new release is needed.

**Review by K1-R1 — APPROVED** [review]

**Pull request merged**